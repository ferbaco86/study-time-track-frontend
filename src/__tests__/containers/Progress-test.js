import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../reducers/store';
import Progress from '../../containers/Progress';

describe('Rendering component', () => {
  it('creates a Progress component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <Progress />
          </Provider>
        </BrowserRouter>,
      );
    });
    screen.getByText('Longest Session');
  });
});

describe('Progress', () => {
  it('renders correctly', () => {
    const snap = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Progress />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
