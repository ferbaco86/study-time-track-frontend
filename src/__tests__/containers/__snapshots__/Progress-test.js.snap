// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress renders correctly 1`] = `
<div>
  <h1
    className="sc-eCssSg gJiiYa"
  />
  <h4
    className="sc-kEjbxe iSVVHw"
  >
    Latest Session
  </h4>
  <h4
    className="sc-kEjbxe iSVVHw"
  >
    Longest Session
  </h4>
  <h4
    className="sc-kEjbxe iSVVHw"
  >
    Top 5 Most Studied Subjects
  </h4>
</div>
`;
